lib/client/backend_client.dart

```diff
@@ -1,23 +1,51 @@
+//import 'dart:convert';
+
+import 'dart:convert';
+
 import 'package:http/http.dart' as http;
 
 class BackendClient {
   const BackendClient();
 
 //functions to connect with API
-
-  Future<dynamic> getData({required String uri}) async {
+  Future<List<Map<String, String>>> getData({required String uri}) async {
```

Comment:
"A" for effort! Being as specific as you can with the type is a great practice.

For the client layer, Although it works for this use case, it would be better to have a `Future<dynamic>` here and then get more specific in the service layer. What if your response is a single object rather than a JSON array? Then you would need `Future<Map<String, String>>`. What if your JSON values were not `String`s? Then you would need `Future<List<Map<String, dynamic>>>`.

When the `Future<dynamic>` gets sent back to the service, the service layer gives you the ability to be very specific about the type.

Reply:
I made a change!